sdk/monitor/monitor-ingestion/review/monitor-ingestion.api.md

```diff
@@ -25,25 +35,9 @@ export interface UploadLogsError {
 }
 
 // @public
-export interface UploadLogsOptions {
+export interface UploadLogsOptions extends OperationOptions {
+    errorCallback?: (uploadLogsError: UploadLogsError) => Promise<void>;
```

Comment:
I think `UploadLogsError` should either be an actual Error class (`extends Error`) or be renamed to something that doesn't look like an Error class, such as `LogsUploadFailure`

Reply:
Also it feels weird to have this be a Promise that is awaited by the system, since usually callbacks are fire and forget, is there a reason that upload should block on the callback?

Reply:
So the `cause` field in the UploadLogsError is the actual `Error`
`export interface UploadLogsError {
  /**
   * List of failed logs
   */
  failedLogs: Record<string, unknown>[];
  /**
   * Error for failed logs
   */
  cause: Error;
}`

Reply:
My concern is that if I see the identifier `UploadLogsError` my brain thinks it is an Error class as I have been trained to assume any type name that follows the convention `PascalCaseWordsError` is an Error and not an object interface for something else.

This naturally creates confusion when UploadLogsError doesn't have a `name` or a `message` or a `stack`

Reply:
I agree with @xirzec, LogsUploadFailure sounds more accurate as the object itself has a failed log, not only the error.